sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -155,6 +159,67 @@ dif_rstmgr_result_t dif_rstmgr_reset_info_clear(const dif_rstmgr_t *handle) {
   return kDifRstmgrOk;
 }
 
+dif_rstmgr_result_t dif_rstmgr_alert_info_set_enabled(
+    const dif_rstmgr_t *handle, dif_rstmgr_toggle_t state) {
+  if (handle == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  uint32_t enabled = (state == kDifRstmgrToggleEnabled) ? 0x1 : 0x0;
+
+  // This will clobber the `ALERT_INFO_CTRL.INDEX` field. However, the index
+  // field is only relevant during the crash dump read operation, and is
+  // set by the caller and not the hardware, so it is safe to clobber it.
+  mmio_region_write32(base_addr, RSTMGR_ALERT_INFO_CTRL_REG_OFFSET, enabled);
+
+  return kDifRstmgrOk;
+}
+
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_get_size(
+    const dif_rstmgr_t *handle, size_t *count) {
+  if (handle == NULL || count == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  *count = rstmgr_alert_info_dump_get_size(base_addr);
+
+  return kDifRstmgrOk;
+}
+
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_read(
+    const dif_rstmgr_t *handle, size_t index,
+    dif_rstmgr_alert_info_dump_value_t *value) {
```

Comment:
Just a suggestion: If 
* `INDEX` in `ALERT_INFO_CTRL` is reset to 0 on system reset, and
* We don't need to support reading the same dump multiple times,

then, we can use `INDEX` in `ALERT_INFO_CTRL` to keep track of the read operation and have something like:
```
result_t dif_rstmgr_alert_info_dump_read(... *handle, ... *buffer, buffer_len, size_t *words_written)
```

But, this doesn't make a whole lot of sense if the only use-case of this API is to dump these values one by one over a serial interface. Something like this might be better:
```
// index >= cnt_avail?
result_t dif_rstmgr_alert_info_dump_is_empty(... *handle, ... *is_empty);
// read next dump word
result_t dif_rstmgr_alert_info_dump_read_next(... *handle, ... *dump_word)
```

Reply:
Actually, probably even better solution would be @lenary suggested `typedef struct dif_rstmgr_info_dump{type data[index_mask_size]}`. 

`result_t dif_rstmgr_alert_info_dump_read(... *handle, ... dif_rstmgr_info_dump_t *dump, size_t *words_written)`

Reply:
I agree, max size of the dump is pretty small. What do you think about putting the size, i.e. the actual number of words, in the struct as well?